README.md

```diff
@@ -114,6 +114,34 @@ dependencies {
 }
 ```
 
+## Development
+### Setup
+* development requires java 11 to be installed
+* development assumes you are running on Ubuntu, while you may be able to use other operating systems they are not supported for developement.
+* before creating commits
+  * read https://www.conventionalcommits.org/en
+  * create the following script in your .git/hooks directory and name it commit.msg
```

Comment:
if we really need a script to help using conventional commit, we can introduce [Commitizen](https://commitizen.github.io/cz-cli/) for better automation.

Reply:
Ok. I based it on the fact that I ran into errors when doing the functional testing in my Windows machine and that we use the Ubuntu image on the CI server. When I get a chance I will update it.